api/turing/validation/validator.go

```diff
@@ -118,6 +120,23 @@ func newExperimentConfigValidator(expSvc service.ExperimentsService) func(valida
 	return validationFunc
 }
 
+func checkTrafficRuleName(sl validator.StructLevel, fieldName string, value string) {
+	nameRegexDescription := strings.Join([]string{
+		"Name must be between 4-64 characters long, and begin with an alphanumeric character",
+		"and have no trailing spaces and can contain letters, numbers, blank spaces and the following symbols: -_()#$&:.",
+	}, " ")
+	if !trafficRuleNameRegex.MatchString(value) {
+		sl.ReportError(value, fieldName, "trafficRuleName", nameRegexDescription, fmt.Sprintf("%v", value))
+	}
+}
+
+func checkTrafficRulesUniqueName(sl validator.StructLevel, fieldName string, value []string) {
```

Comment:
Can we replace the custom uniqueness check with the tag: https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Unique

Reply:
Would you mean placing `"validate:unique=name"` tag for `TrafficRules` here - https://github.com/caraml-dev/turing/blob/main/api/turing/models/traffic_rules.go#L16?

Reply:
I think it needs to be added in the parent struct ([here](https://github.com/caraml-dev/turing/blob/main/api/turing/api/request/request.go#L27)). But I haven't used it in the past, so please help review the docs / test it out.

Reply:
Indeed, we need to add to the parent struct, more specifically, `unique=Name` should come before `dive`.

Reply:
This looks good, thanks!